src/Microsoft.Management.Deployment/PackageManager.cpp

```diff
@@ -422,6 +426,30 @@ namespace winrt::Microsoft::Management::Deployment::implementation
             {
                 Microsoft::Management::Deployment::PackageVersionInfo packageVersionInfo = GetPackageVersionInfo(package, options);
                 std::unique_ptr<COMContext> comContext = CreateContextFromInstallOptions(package, options, callerProcessInfoString);
+
+                if (isUpgrade)
+                {
+                    AppInstaller::Utility::VersionAndChannel installedVersion{ winrt::to_string(package.InstalledVersion().Version()), winrt::to_string(package.InstalledVersion().Channel()) };
+                    AppInstaller::Utility::VersionAndChannel upgradeVersion{ winrt::to_string(packageVersionInfo.Version()), winrt::to_string(packageVersionInfo.Channel()) };
+
+                    if (installedVersion.IsUpdatedBy(upgradeVersion) ||
+                        (options.AllowUpgradeToUnknownVersion() &&
+                         AppInstaller::Utility::ICUCaseInsensitiveEquals(installedVersion.GetChannel().ToString(), upgradeVersion.GetChannel().ToString()) &&
+                         upgradeVersion.GetVersion().IsUnknown()))
+                    {
+                        // Set upgrade flag
+                        comContext->SetFlags(AppInstaller::CLI::Execution::ContextFlag::InstallerExecutionUseUpdate);
+                        // Add installed version
+                        winrt::Microsoft::Management::Deployment::implementation::PackageVersionInfo* installedVersionInfoImpl = get_self<winrt::Microsoft::Management::Deployment::implementation::PackageVersionInfo>(package.InstalledVersion());
+                        std::shared_ptr<::AppInstaller::Repository::IPackageVersion> internalInstalledVersion = installedVersionInfoImpl->GetRepositoryPackageVersion();
+                        comContext->Add<AppInstaller::CLI::Execution::Data::InstalledPackageVersion>(internalInstalledVersion);
+                    }
+                    else
+                    {
+                        co_return GetInstallResult(executionStage, APPINSTALLER_CLI_ERROR_UPDATE_NOT_APPLICABLE, correlationData, false);

```

Comment:
I think that we might need to break down the specific cases more for these errors.  Specifically, I'm concerned about overloading this error (based on what I think is returning it, so I could be wrong).  If the update does not have an applicable installer, do we get this error?

If the caller gives us a version that is less than or equal to the current version, should we have a different error?  If the caller does not give us a version, but the latest version is already installed, what error will be produced, if any?